File: src/c5/BrowserInsult.java
package c5;

import java.io.IOException;
import java.io.PrintWriter;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import util.ServletUtilities;

public class BrowserInsult extends HttpServlet {
	public void doGet(HttpServletRequest req, HttpServletResponse res) 
			throws ServletException, IOException {
		res.setContentType("text/html");
		PrintWriter out = res.getWriter();
		String title, message;
		String userAgent = req.getHeader("User-Agent");
		if ((userAgent != null) && (userAgent.indexOf("MSIE") != -1)) {
			title = "Microsoft Minion";
			message = "Welcome, O spineless slave to the mighty empire.";
		} else {
			title = "Hopeless Netscape Rebel";
			message = "Enjoy it while you can. You <i>will</i> be assimilated!";
		}
		out.println(ServletUtilities.headWithTitle(title) +
				"<body bgcolor=\"#FDF5E6\">\n" +
				"<h1 align=\"center\">" + title + "</h1>\n" +
				message + "\n" +
				"</body></html>");
	}
}

File: src/gocode/c5/browser_insult.go
package c5

import (
	"fmt"
	"net/http"
	"strings"

	"../util"
)

type BrowserInsult struct {
}

func (bi *BrowserInsult) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	var err error
	switch r.Method {
	case "GET":
		err = bi.handleGet(w, r)
	case "POST":
		err = bi.handlePost(w, r)
	default:
		err = fmt.Errorf("%v", r.Method+" may not be supported")
	}
	if err != nil {
		http.Error(w, err.Error(), http.StatusInternalServerError)
		return
	}
}

func (bi *BrowserInsult) handleGet(w http.ResponseWriter, r *http.Request) (err error) {
	w.Header().Set("Content-Type", "text/html")
	var title, msg string
	if strings.Index(r.UserAgent(), "MSIE") != -1 {
		title = "Microsoft Minion"
		msg = "Welcome, O spineless slave to the mighty empire."
	} else {
		title = "Hopeless Netscape Rebel"
		msg = "Enjoy it while you can. You <i>will</i> be assimilated!"
	}
	w.Write([]byte(util.HeadWithTitle(title) +
		"<body bgcolor=\"#FDF5E6\">\n" +
		"<h1 align=\"center\">" + title + "</h1>\n" +
		msg + "\n" +
		"</body></html>"))

	return
}

func (bi *BrowserInsult) handlePost(w http.ResponseWriter, r *http.Request) (err error) {
	err = bi.handleGet(w, r)

	return
}
